code/modules/art/paintings.dm

```diff
@@ -503,6 +503,104 @@
 	result_path = /obj/structure/sign/painting
 	pixel_shift = 30
 
+/obj/item/wallframe/painting/syndicate_teleporter
+	result_path = /obj/structure/sign/painting/syndicate_teleporter
+
+/obj/structure/sign/painting/syndicate_teleporter
+	damage_deflection = 20
+	///Where a human gets teleported to once they activate the teleporter, picks from a list of GLOB.active_syndicate_beacons
+	var/atom/teleport_target
+	///If we've taken enough damage, we become unusable
+	var/integrity_compromised = FALSE
+
+/obj/structure/sign/painting/syndicate_teleporter/Initialize(mapload, dir, building)
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_TAKE_DAMAGE, PROC_REF(on_damage_taken))
+	RegisterSignals(src, list(COMSIG_QDELETING,	COMSIG_MACHINERY_BROKEN, TELEPORTER_SET_TARGET), PROC_REF(remove_connections))
+
+/obj/structure/sign/painting/syndicate_teleporter/Destroy()
+	if(current_canvas)
+		current_canvas.forceMove(drop_location())
```

Comment:
No side effects in destroy. This should be `QDEL_NULL(current_canvas)`, and you should implement `/obj/structure/sign/painting/syndicate_teleporter/atom_deconstruct()` which knocks it off the wall